TACC-Cloud/hazmapper
# Bug/DES-2184: Add way to select overlapping layers.

## Overview: ##
Implement a way to select point cloud layers that are overlapping through a context menu.

## PR Status: ##


## Related Jira tickets: ##

* [DES-2184](https://jira.tacc.utexas.edu/browse/DES-2184)

## Summary of Changes: ##
#### Asset styling
Added indication for active assets (red) and added default styles for assets (as constants). I also added extra asset types in `leafletUtils.js` to include style default asset types. Also lowered opacity of `Polygon` assets as per Michael's request.

#### Handling overlapping layers
For now, I put `Polygon` assets in a lower z-index compared to the `Point` assets just in case that some `Point` assets are unselectable.

I installed `leaflet-contextmenu` ([link](https://github.com/aratcliffe/Leaflet.contextmenu)) and added basic type definitions to get rid of errors.
I use this `contextmenu` to give the user a menu based on if there are overlapping `Polygon` layers in on the pointer position. If there is only one `Polygon`, we just select it.

The rendered asset at the top (highest implied `z-index`) should be at the top of the `contextmenu`.

#### Asset labels
As Michael suggested, I put on optional/toggleable labels for `Polygon` assets for clarification. I haven't figured out how we would use the Point cloud description for the names rather than just the `featureId`.

## Testing Steps: ##
1. Ensure that asset label toggling works for `Polygon` assets.
2. Ensure that overlapping assets are correctly selected with the `contextmenu`.
3. Ensure that selected assets (both `Point` and `Polygon`) change color on selection.

## UI Photos:
<img width="1265" alt="Screen Shot 2022-03-09 at 3 02 15 PM" src="https://user-images.githubusercontent.com/9425579/157538664-7a2f727f-8aa7-4860-acdf-f299792219e3.png">

## Notes: ##
